### DISCUSSION TITLE
the recovery time of  data transmission  is too long

### DISCUSSION THREAD

**Discussion by FantasticSaltedFish** [eProsima/Fast-DDS#3724] [Unofficial support]
### Is there an already existing issue for this?

- [X] I have searched the existing issues

### Expected behavior

After the network is disconnected, I expect fastdds could quickly resume data transmission within 10 seconds

### Current behavior

There is an experiment where the HelloWorldExample's publisher is sending data to the HelloWorldExample's subscriber and everything is well. However, when unplugging the internet cable and then inserting it again (about 1 second), the subscriber needs to wait for about 30 seconds to resume receiving the publisher's data. The disconnection time for this data transmission is too long. Is there any way to shorten the recovery time for this transmission?

### Steps to reproduce

HelloWorldExample program, set count=1000 and sleep=1000ms. After compiling the program, copy it to two computers separately;

Computer A runs the subscription program, and then Computer B runs the publishing program;

During the communication process between the publishing program and the subscribing program, unplug the internet cable and immediately plug it in (approximately 1 second);

It is found that the communication recovery between the publisher and the subscriber takes approximately 30 seconds.

### Fast DDS version/commit

fastdds V2.3

### Platform/Architecture

Windows 10 Visual Studio 2019

### Transport layer

UDPv4

### Additional context

no

### XML configuration file

```xml
no
```

### Relevant log output

```shell
no
```

### Network traffic capture

no

**Comment by JLBuenoLopez** [comment]
Thanks for your contribution @FantasticSaltedFish,

First of all, Fast DDS v2.3 is no longer supported so I encourage that you upgrade to one of the latest Fast DDS versions.
Maybe the issue has already been fixed.

I understand that you are not modifying the QoS of the example. The [Liveliness QoS Policy](https://fast-dds.docs.eprosima.com/en/latest/fastdds/dds_layer/core/policy/standardQosPolicies.html#livelinessqospolicy) might affect the reconnection but if the default configuration is used, then it should not affect. It would be nice to have a wireshark capture to see the traffic being exchanged among the applications.

I am moving this to the non-officially supported forum as Fast DDS v2.3 is not alive. Please, consider opening a new ticket if you confirm that the issue is still in a supported Fast DDS version and consider attaching a Wireshark capture (a mail can be provided so you can send it privately if needed).
